filament/src/MaterialInstanceManager.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <utils/bitset.h>
+
+#include <vector>
+
+namespace filament {
+
+class FMaterial;
+class FMaterialInstance;
+class FEngine;
+
+// This class manages a cache of material instances mapped by materials. Having a cache allows us to
+// re-use instances across frames.
+class MaterialInstanceManager {
+public:
+    class Record {
+    public:
+        Record(FMaterial const* material)
+            : mMaterial(material) {}
+        ~Record() = default;
+
+        Record(Record const& rhs) noexcept = default;
+        Record& operator=(Record const& rhs) noexcept = default;
+        Record(Record&& rhs) noexcept = default;
+        Record& operator=(Record&& rhs) noexcept = default;
```

Comment:
all these ctor/dtors are inline. is it what you wanted? This will inline the std::vector construction/destruction. I think these should probably not be inline.

Reply:
moved to cpp